### PR TITLE
Ifpack2: More changes for BlockTriDi example

### PR DESCRIPTION
**Pull request by csiefer2**
This now correctly works with my single-processor version of the dump from the app.

Issue: #11195 
Stakeholder feedback: n/a
Tests: This is one.

### CODE REVIEW

**Review by kliegeois — COMMENTED** [review]
Thanks a lot @csiefer2 ! Just couple of small comments.

File: packages/ifpack2/example/BlockTriDi.cpp [hunk 1]
```diff
@@ -347,21 +379,29 @@ main (int argc, char* argv[])
...
 
     // Assume contiguous blocks here
-    for(LO i=0; i<(LO)line_ids.size(); i+=args.blockSize) {
-      LO block_lid = i / args.blockSize;
+    for(LO i=0; i<(LO)line_ids.size(); i++) {
+      LO block_lid = i;
       LO block_num = line_ids[i];
       parts[block_num].push_back(block_lid);     
     }      
+    std::cout<<"On "<<line_ids.size()<<" local DOFs, detected "<<num_local_lines<<" lines"<<std::endl;
```
[the remark this answers is not in the document]
Comment (csiefer2) [reply packages/ifpack2/example/BlockTriDi.cpp]: Good call. I just commented that out.

**Review by kliegeois — APPROVED** [review]
Thanks @csiefer2 !

**Review by kliegeois — APPROVED** [review]
I tried the branch and it works on weaver!

**Comment by sebrowne** [comment]
I took a brief look at the failed CUDA test, it looks like the test passed but the output got interleaved due to a buffering issue or some other such trickery and the test harness couldn't find the "TEST PASSED" string (not 100% sure that's what it is, but 99%).  FWIW

**Comment by csiefer2** [comment]
@sebrowne Ross might hate me for it, but this is a "Slap on AT: RETEST and try again sort of moment"

**Comment by rppawlo** [comment]
@csiefer2 - I would disagree. The test needs to be fixed such that only a single mpi process outputs all output. This prevents the pass string from getting mangled by other mpi process output. Or the test should use a different metric for determining if it passed. Otherwise this random failure will happen again.

**Comment by csiefer2** [comment]
@rppawlo I agree that test needs to be changed.  But it is not the job of my PR --- which doesn't touch Piro --- to change it.

Perhaps we should open up an issue with the Piro team to fix the test...

**Comment by sebrowne** [comment]
This time it's another one of these seemingly nondeterministic "address not mapped" errors that I've been observing with the CUDA lines in particular.  I have work scheduled this week that I think will alleviate this (we think it's related to overloading a single GPU on the test machine), but I'm unsure of a good approach to immediately get the PRs through.  In the short term, we could reinstate the list of tests that were disabled for CUDA10 (see below), but I'm not sure that is a good idea long-term.


Adelus_vector_random_MPI_1_DISABLE
Adelus_vector_random_MPI_2_DISABLE
Adelus_vector_random_MPI_3_DISABLE
Adelus_vector_random_MPI_4_DISABLE
Amesos_Test_Detailed_MPI_4_DISABLE
Amesos_Test_LAPACK_MPI_4_DISABLE
Amesos_compare_solvers_MPI_4_DISABLE
Anasazi_Epetra_IRTR_auxtest_0_MPI_4_DISABLE
Anasazi_Epetra_IRTR_auxtest_1_MPI_4_DISABLE
Anasazi_Epetra_IRTR_auxtest_2_MPI_4_DISABLE
Anasazi_Epetra_IRTR_auxtest_3_MPI_4_DISABLE
Anasazi_Epetra_IRTR_auxtest_4_MPI_4_DISABLE
Anasazi_Epetra_LOBPCG_auxtest_MPI_4_DISABLE
Anasazi_Epetra_LOBPCG_solvertest_MPI_4_DISABLE
Belos_bl_fgmres_hb_MPI_4_DISABLE
Belos_bl_pgmres_hb_0_MPI_4_DISABLE
EpetraExt_inout_test_LL_MPI_4_DISABLE
EpetraExt_inout_test_MPI_4_DISABLE
FEI_lagrange_20quad_old_MPI_1_DISABLE
Ifpack_Container_MPI_4_DISABLE
Ifpack_Container_LL_MPI_4_DISABLE
Ifpack2_AdditiveSchwarz_MPI_4_DISABLE
Intrepid2_unit-test_Discretization_Basis_DeRHAM_TET_FEM_test_01_Serial_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_DeRHAM_TRI_FEM_test_01_Serial_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HCURL_TET_In_FEM_test_01_CUDA_DFAD_DFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HCURL_TET_In_FEM_test_01_CUDA_DOUBLE_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HCURL_TET_In_FEM_test_01_CUDA_SFAD_SFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HCURL_TET_In_FEM_test_01_CUDA_SLFAD_DFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HCURL_TET_In_FEM_test_01_CUDA_SLFAD_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HCURL_TET_In_FEM_test_01_Serial_DFAD_DFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HCURL_TET_In_FEM_test_01_Serial_DOUBLE_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HCURL_TET_In_FEM_test_01_Serial_SFAD_SFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HCURL_TET_In_FEM_test_01_Serial_SLFAD_DFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HCURL_TET_In_FEM_test_01_Serial_SLFAD_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HDIV_TET_In_FEM_test_01_CUDA_DFAD_DFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HDIV_TET_In_FEM_test_01_CUDA_DOUBLE_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HDIV_TET_In_FEM_test_01_CUDA_SFAD_SFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HDIV_TET_In_FEM_test_01_CUDA_SLFAD_DFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HDIV_TET_In_FEM_test_01_CUDA_SLFAD_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HDIV_TET_In_FEM_test_01_Serial_DFAD_DFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HDIV_TET_In_FEM_test_01_Serial_DOUBLE_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HDIV_TET_In_FEM_test_01_Serial_SFAD_SFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HDIV_TET_In_FEM_test_01_Serial_SLFAD_DFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HDIV_TET_In_FEM_test_01_Serial_SLFAD_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HGRAD_TRI_Cn_FEM_test_01_CUDA_DFAD_DFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HGRAD_TRI_Cn_FEM_test_01_CUDA_DOUBLE_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HGRAD_TRI_Cn_FEM_test_01_CUDA_SFAD_SFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HGRAD_TRI_Cn_FEM_test_01_CUDA_SLFAD_DFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HGRAD_TRI_Cn_FEM_test_01_CUDA_SLFAD_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HGRAD_TRI_Cn_FEM_test_01_Serial_DFAD_DFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HGRAD_TRI_Cn_FEM_test_01_Serial_DOUBLE_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HGRAD_TRI_Cn_FEM_test_01_Serial_SFAD_SFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HGRAD_TRI_Cn_FEM_test_01_Serial_SLFAD_DFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HGRAD_TRI_Cn_FEM_test_01_Serial_SLFAD_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HVOL_TRI_Cn_FEM_test_01_CUDA_DFAD_DFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HVOL_TRI_Cn_FEM_test_01_CUDA_DOUBLE_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HVOL_TRI_Cn_FEM_test_01_CUDA_SFAD_SFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HVOL_TRI_Cn_FEM_test_01_CUDA_SLFAD_DFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HVOL_TRI_Cn_FEM_test_01_CUDA_SLFAD_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HVOL_TRI_Cn_FEM_test_01_Serial_DFAD_DFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HVOL_TRI_Cn_FEM_test_01_Serial_DOUBLE_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HVOL_TRI_Cn_FEM_test_01_Serial_SFAD_SFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HVOL_TRI_Cn_FEM_test_01_Serial_SLFAD_DFAD_MPI_1_DISABLE
Intrepid2_unit-test_Discretization_Basis_HVOL_TRI_Cn_FEM_test_01_Serial_SLFAD_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_MonolithicExecutable_Intrepid2_Tests_MPI_1_DISABLE
Intrepid2_unit-test_Orientation_test_orientation_HEX_Serial_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Orientation_test_orientation_HEX_newBasis_Serial_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Orientation_test_orientation_TET_Serial_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Projection_test_DeRham_commutativity_TET_CUDA_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Projection_test_DeRham_commutativity_TET_Serial_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Projection_test_convergence_TET_CUDA_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Projection_test_convergence_TET_Serial_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Projection_test_interpolation_projection_HEX_CUDA_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Projection_test_interpolation_projection_HEX_Serial_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Projection_test_interpolation_projection_TET_CUDA_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Projection_test_interpolation_projection_TET_Serial_DOUBLE_MPI_1_DISABLE
Intrepid_test_Discretization_Basis_HCURL_HEX_In_FEM_Test_02_MPI_1_DISABLE
Intrepid_test_Discretization_Basis_HCURL_TET_In_FEM_Test_02_MPI_1_DISABLE
Intrepid_test_Discretization_Basis_HDIV_HEX_In_FEM_Test_02_MPI_1_DISABLE
Intrepid_test_Discretization_Basis_HDIV_QUAD_In_FEM_Test_02_MPI_1_DISABLE
Intrepid_test_Discretization_Basis_HDIV_TET_In_FEM_Test_02_MPI_1_DISABLE
Intrepid_test_Discretization_Basis_HDIV_TRI_In_FEM_Test_02_MPI_1_DISABLE
Intrepid_test_Discretization_Basis_HGRAD_HEX_Cn_FEM_Test_02_MPI_1_DISABLE
Intrepid_test_Discretization_Basis_HGRAD_LINE_Hermite_FEM_Test_01_MPI_1_DISABLE
Intrepid_test_Discretization_Basis_HGRAD_LINE_Hermite_FEM_Test_02_MPI_1_DISABLE
Intrepid_test_Discretization_Basis_HGRAD_TET_Cn_FEM_Test_02_MPI_1_DISABLE
Intrepid_test_Discretization_Basis_HGRAD_TRI_Cn_FEM_ORTH_Test_02_MPI_1_DISABLE
Intrepid_test_Discretization_Basis_HGRAD_TRI_Cn_FEM_Test_01_MPI_1_DISABLE
Intrepid_test_Discretization_Basis_HGRAD_TRI_Cn_FEM_Test_02_MPI_1_DISABLE
KokkosKernels_blas_serial_MPI_1_DISABLE
MiniTensor_test_Test_01_MPI_1_DISABLE
MiniTensor_test_Test_02_MPI_1_DISABLE
MueLu_UnitTestsIntrepid2Tpetra_MPI_4_DISABLE
MueLu_BlockSmoothingWithAverages_MPI_4_DISABLE BOOL :ON
MueLu_DriverDiagonalModifications_MPI_1_DISABLE BOOL :ON
MueLu_GeneralBlockSmoothing_MPI_4_DISABLE BOOL :ON
MueLu_UnitTestsTpetra_kokkos_MPI_1_DISABLE BOOL :ON
MueLu_UnitTestsTpetra_kokkos_MPI_4_DISABLE BOOL :ON
PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4_DISABLE
PanzerAdaptersSTK_MixedCurlLaplacianExample-ConvTest-Tri-Order-1_DISABLE
PanzerAdaptersSTK_MixedPoissonExample-ConvTest-Hex-Order-3_DISABLE
PanzerAdaptersSTK_PoissonInterfaceExample_3d_MPI_4_DISABLE
PanzerAdaptersSTK_main_driver_energy-ss-blocked-tp_DISABLE
PanzerDofMgr_Test_FE_Assembly_HEX_MPI_2_DISABLE
PanzerMiniEM_MiniEM-BlockPrec_Augmentation_MPI_4_DISABLE
PanzerMiniEM_MiniEM-BlockPrec_RefMaxwell_MPI_4_DISABLE
Piro_AnalysisDriverTpetra_MPI_4_DISABLE
Piro_ThyraSolverTpetra_MPI_4_DISABLE
ROL_example_PDE-OPT_0ld_adv-diff-react_example_01_MPI_4_DISABLE
ROL_example_PDE-OPT_0ld_adv-diff-react_example_02_MPI_4_DISABLE
ROL_example_PDE-OPT_0ld_poisson_example_01_MPI_4_DISABLE
ROL_example_PDE-OPT_0ld_stefan-boltzmann_example_03_MPI_4_DISABLE
ROL_example_PDE-OPT_navier-stokes_example_01_MPI_4_DISABLE
ROL_example_PDE-OPT_navier-stokes_example_02_MPI_4_DISABLE
ROL_example_PDE-OPT_nonlinear-elliptic_example_01_MPI_4_DISABLE
ROL_example_PDE-OPT_nonlinear-elliptic_example_02_MPI_4_DISABLE
ROL_example_PDE-OPT_obstacle_example_01_MPI_4_DISABLE
ROL_example_PDE-OPT_poisson-boltzmann_example_01_MPI_4_DISABLE
ROL_example_PDE-OPT_topo-opt_poisson_example_01_MPI_4_DISABLE
ROL_example_poisson-inversion_example_01_MPI_1_DISABLE
ROL_test_elementwise_TpetraMultiVector_MPI_4_DISABLE
ROL_test_function_sketching_Interface_MPI_1_DISABLE
SEACASIoss_exodus32_to_exodus32_DISABLE
SEACASIoss_exodus_fpp_serialize_DISABLE
Stratimikos_test_amesos_thyra_driver_MPI_1_DISABLE
Stratimikos_test_single_amesos2_tpetra_solver_driver_KLU2_MPI_1_DISABLE
Teko_testdriver_MPI_4_DISABLE
Teko_DiagnosticLinearOp_test_MPI_1_DISABLE
TeuchosNumerics_BLAS_test_MPI_1_DISABLE
ThyraTpetraAdapters_Simple2DTpetraModelEvaluatorUnitTests_MPI_1_DISABLE
ThyraTpetraAdapters_TpetraThyraWrappersUnitTests_MPI_4_DISABLE
ThyraTpetraAdapters_TpetraThyraWrappersUnitTests_serial_MPI_1_DISABLE
TpetraTSQR_FullTsqr_Accuracy_5000rows_100cols_Sequential_MPI_4_DISABLE
Zoltan2_fix4785_MPI_4_DISABLE
Pliris_vector_random_MPI_3_DISABLE
Pliris_vector_random_MPI_4_DISABLE
Intrepid2_unit-test_Projection_test_DeRham_commutativity_HEX_CUDA_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Projection_test_DeRham_commutativity_QUAD_CUDA_DOUBLE_MPI_1_DISABLE
Intrepid2_unit-test_Projection_test_DeRham_commutativity_TRI_CUDA_DOUBLE_MPI_1_DISABLE

**Pull request merged**